city_scrapers/spiders/il_arts_council.py

```diff
@@ -0,0 +1,129 @@
+                    classification=self._parse_classification(item),
+                    start=self._parse_start(item, year),
+                    end=self._parse_end(item),
+                    all_day=self._parse_all_day(item),
+                    time_notes=self._parse_time_notes(item),
+                    location=self._parse_location(item),
+                    links=self._parse_links(item),
+                    source=self._parse_source(response),
+                )
+
+                meeting_status = item.xpath('td[2]/text()').get() or ""
+                meeting["status"] = self._get_status(meeting, text=meeting_status)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        return "Agency Board"
+
+    def _parse_description(self, item):
+        """Parse or generate meeting description."""
+        return ""
+
+    def _parse_classification(self, item):
+        """Parse or generate classification from allowed options."""
+        return BOARD
+
+    def _parse_start(self, item, year):
+        """Parse start datetime as a naive datetime object."""
+        item_text = item.xpath('./td/text()').get()
+        if not item_text:
+            item_text = item.xpath('./td/a/text()').get()
+        text_clean = unicodedata.normalize("NFKD", item_text).strip()
+
+        if text_clean[-4::] == year:
+            start = dt.strptime(text_clean, '%A, %B %d, %Y')
+        elif text_clean[-2::] in ['am', 'pm']:
+            start = dt.strptime(text_clean + year, '%A, %B %d, %I:%M%p%Y')
+        else:
+            start = dt.strptime(text_clean + year, '%A, %B %d%Y')
+
+        return start
+
+    def _parse_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        return None
+
+    def _parse_time_notes(self, item):
+        """Parse any additional notes on the timing of the meeting"""
+        return ""
+
+    def _parse_all_day(self, item):
+        """Parse or generate all-day status. Defaults to False."""
+        return False
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+
+        location = item.xpath('td[2]/text()').get()
+        if not location:
+            location = item.xpath('td[2]/a/text()').get()
+
+        jrtc = {
+            "address": "100 West Randolph, Suite 10-500, Chicago, IL 60601",
+            "name": "IACA/JRTC"
+        }
+
+        tba = {
+            "address": "TBA",
+            "name": "TBA",
+        }
+
+        if 'TBA' in location:
+            return tba
+
+        elif 'JRTC' in location:
+            return jrtc
+
+        else:
+            return {
+                "address": "",
+                "name": "",
+            }
+
+    def _parse_links(self, item):
+        """Parse or generate links."""
+        agenda_link = item.xpath('td/a/@href').get()
```

Comment:
I missed this on the first pass, but it looks like some of the URLs being returned are relative (seeing it on the August 2018 meeting). To be safe, we should add `response` as an argument to this function and call `response.urljoin` on the parsed link to make sure that it includes the domain

Reply:
Good catch, thanks.